### PR TITLE
Add error handling in server

### PR DESCRIPTION
**Pull request by karenfrilya97**
Added error handlers which:
- Handles error occurring in the server and 
- Returns appropriate error status code and message to the client

Merge after PR #32

### CODE REVIEW

File: server/src/api/merchants/index.ts [hunk 1]
```diff
@@ -24,9 +24,19 @@ import {merchantsService} from 'services';
...
-});
+router.get(
+  '/',
+  async (
+    req: express.Request,
+    res: express.Response,
+    next: express.NextFunction
+  ) => {
+    try {
+      const merchants = (await merchantsService.getMerchants())[0];
```
Comment (ifhuang) [inline server/src/api/merchants/index.ts:35]: Why you don't set res anymore?

Comment (karenfrilya97) [reply server/src/api/merchants/index.ts:35]: ohh thanks for noticing! I accidentally removed the `res.send()`

File: server/src/index.ts [hunk 2]
```diff
@@ -35,7 +35,34 @@ router.use('/customers', customersRouter);
 router.use('/listings', listingsRouter);
 router.use('/merchants', merchantsRouter);
 
-app.use('/', router);
+app.use(router);
+
+// handles server error
```
Comment (chshda) [inline server/src/index.ts:40]: Prefer writing comments with complete sentence, i.e. capital first letter and full stop at the end.

Same as the rest.

Comment (karenfrilya97) [reply server/src/index.ts]: okay!

**Pull request merged**